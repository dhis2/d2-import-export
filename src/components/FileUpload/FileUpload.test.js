import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect'

import { FileUpload } from '.'

const props = {
    dataTest: 'file-upload',
    name: 'file',
    setFile: () => 1,
}

it('matches snapshot', () => {
    const { asFragment } = render(<FileUpload {...props} />)
    expect(asFragment()).toMatchSnapshot()
})

it('shows instructions', () => {
    const { getByDataTest } = render(<FileUpload {...props} />)
    const button = getByDataTest('file-upload-fileinput')
    expect(button).toHaveTextContent('Choose a file to upload')
})

it('shows filename', () => {
    const file = new File([''], 'testfile.json', { type: 'application/json' })
    const { getByDataTest } = render(<FileUpload {...props} file={file} />)
    const button = getByDataTest('file-upload-fileinput')
    expect(button).toHaveTextContent('testfile.json')
})
