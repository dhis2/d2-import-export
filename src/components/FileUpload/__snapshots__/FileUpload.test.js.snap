// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="file-upload"
  >
    <span
      class="label"
    >
      File
    </span>
    <div
      class="jsx-1051217026 file-input"
      data-test="file-upload-fileinput"
    >
      <input
        accept="*"
        class="jsx-1051217026"
        id="file"
        name="file"
        type="file"
      />
      <button
        class="jsx-3433386394 icon"
        data-test="dhis2-uicore-button"
        type="button"
      >
        <span
          class="jsx-3433386394 button-icon"
        >
          <svg
            class="jsx-4098137027 "
          >
            <path
              class="jsx-4098137027"
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              class="jsx-4098137027"
              d="M9 16h6v-6h4l-7-7-7 7h4zm-4 2h14v2H5z"
            />
          </svg>
        </span>
        Choose a file to upload
      </button>
    </div>
  </div>
</DocumentFragment>
`;
